### ISSUE TITLE
Knowledge distillation on Transformer

### ISSUE THREAD

**Issue by sugeeth14** [pytorch/fairseq#570]
Hi,
I trained a transformer model for English to German translation for using the instructions presented [here](https://github.com/pytorch/fairseq/tree/master/examples/translation#prepare-wmt14en2desh). Now I want to train a smaller model using Knowledge distillation mentioned in this [paper](https://arxiv.org/pdf/1606.07947.pdf). Is such thing supported in Fairseq if not how do I get the logits or soft targets from my model so that I can train a smaller model and is this possible in my case. If any one has tried KD on transformer any ideas or suggestions are welcome.
Thank you.

**Comment by spprabhu** [comment]
Hi @Raghava14  were you able to obtain a KD model for En-De.If yes how.

**Comment by Ir1d** [comment]
Hi, I'm trying to use KD for en-de translation. In [the doc](https://github.com/pytorch/fairseq/tree/master/examples/nonautoregressive_translation#knowledge-distillation) it said `decode the training set to produce a distillation dataset`. Could you give me some hint on how to obtain this distillation dataset after I train my model?

**Comment by RamoramaInteractive** [comment]
Hi @Ir1d, this is exactly the same problem I've encountered. There's no hint how to decode the training set to produce a distillation sataset.
